src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,373 @@
+        /// The method.
+        /// </param>
+        /// <param name="now">
+        /// The now.
+        /// </param>
+        /// <param name="request">
+        /// The request.
+        /// </param>
+        /// <param name="ifMatch">
+        /// The if match.
+        /// </param>
+        /// <param name="md5">
+        /// The md 5.
+        /// </param>
+        /// <param name="size">
+        /// The size.
+        /// </param>
+        /// <param name="contentType">
+        /// The content type.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string AuthorizationHeader(
+            string storageAccount, 
+            string storageKey, 
+            string method, 
+            DateTime now, 
+            HttpWebRequest request, 
+            string ifMatch = "", 
+            string md5 = "", 
+            string size = "", 
+            string contentType = "")
+        {
+            var stringToSign = string.Format(
+                "{0}\n\n\n{1}\n{5}\n{6}\n\n\n{2}\n\n\n\n{3}{4}", 
+                method, 
+                (size == string.Empty) ? string.Empty : size, 
+                ifMatch, 
+                GetCanonicalizedHeaders(request), 
+                GetCanonicalizedResource(request.RequestUri, storageAccount), 
+                md5, 
+                contentType);
+
+            var signatureBytes = Encoding.UTF8.GetBytes(stringToSign);
+            var hmacsha256 = new HMACSHA256(Convert.FromBase64String(storageKey));
+            var authorizationHeader = "SharedKey " + storageAccount + ":"
+                                      + Convert.ToBase64String(hmacsha256.ComputeHash(signatureBytes));
+            return authorizationHeader;
+        }
+
+        /// <summary>
+        /// The create container sas token.
+        /// </summary>
+        /// <param name="accountName">
+        /// The account name.
+        /// </param>
+        /// <param name="containerName">
+        /// The container name.
+        /// </param>
+        /// <param name="key">
+        /// The key.
+        /// </param>
+        /// <param name="accessType">
+        /// The access type.
+        /// </param>
+        /// <param name="validityTimeInDays">
+        /// The validity time in days.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string CreateContainerSasToken(
+            string accountName, 
+            string containerName, 
+            string key, 
+            SasAccessType accessType, 
+            int validityTimeInDays)
+        {
+            var signedpermissions = string.Empty;
+            if (accessType == SasAccessType.read)
+            {
+                signedpermissions += "r";
+            }
+
+            if (accessType == SasAccessType.write)
+            {
+                signedpermissions += "wdl";
+            }
+
```

Comment:
In the future you add a permission, FlibbertyGibbet.  Someone builds using that, and since it's an enum in their program it just says "2".   They then end up using this library, and pass (effectively) `(SasAccessType)2`.  This sets signedPermissions to `string.Empty`.  Is that desired?  Or, do you want

```c#
switch (accessType)
{
    case SasAccessType.Read:
        signedPermissions = "r";
        break;
    case SasAccessType.Write:
       signedPermissions = "wdl";
       break;
    default:
        throw new ArgumentOutOfRangeException(nameof(accessType), accessType, "Unrecognized value");
}
```
?